src/treetensornetworks/solvers/update_step.jl

```diff
@@ -200,38 +184,47 @@ function region_update(
     which_sweep,
     sweep_plan,
     which_region_update,
-    region_kwargs,
     updater_kwargs,
+    internal_kwargs,
   )  # args passed by reference are supposed to be modified out of place
   state = state![] # dereference
   projected_operator = projected_operator![]
   if !(phi isa ITensor && info isa NamedTuple)
     println("Solver returned the following types: $(typeof(phi)), $(typeof(info))")
     error("In alternating_update, solver must return an ITensor and a NamedTuple")
   end
-  normalize && (phi /= norm(phi))
-
-  drho = nothing
-  ortho = "left"    #i guess with respect to ordered vertices that's valid but may be cleaner to use next_region logic
+  #haskey(region_kwargs,:normalize) && ( region_kwargs.normalize && (phi /= norm(phi)) )
+  # ToDo: implement noise term as updater
+  #drho = nothing
+  #ortho = "left"    #i guess with respect to ordered vertices that's valid but may be cleaner to use next_region logic
   #if noise > 0.0 && isforward(direction)
   #  drho = noise * noiseterm(PH, phi, ortho) # TODO: actually implement this for trees...
   # so noiseterm is a solver
   #end
 
-  state, spec = insert_local_tensor(
+  state, spec = insert_local_tensor(state, phi, region; inserter_kwargs..., internal_kwargs)
+
+  !haskey(region_kwargs, :region_printer) && (printer = default_region_update_printer)
+  # only perform update! if region_observer actually passed as kwarg
+  haskey(region_kwargs, :region_observer) && update!(
+    region_observer!;
+    cutoff,
+    maxdim,
+    mindim,
+    which_region_update,
+    sweep_plan,
+    total_sweep_steps=length(sweep_plan),
+    end_of_sweep=(which_region_update == length(sweep_plan)),
     state,
-    phi,
-    region;
-    eigen_perturbation=drho,
-    ortho,
-    normalize,
-    maxdim=region_kwargs.maxdim,
-    mindim=region_kwargs.mindim,
-    cutoff=region_kwargs.cutoff,
+    region,
+    which_sweep,
+    spec,
+    outputlevel,
+    info...,
+    region_kwargs...,
   )
 
-  update!(
-    region_observer!;
+  printer(;
```

Comment:
Call this `region_printer`, pass down from the top level functions `alternating_update`, `tdvp`, etc.